libobs/graphics/graphics-ffmpeg.c

```diff
@@ -148,6 +148,55 @@ static void *ffmpeg_image_copy_data_straight(struct ffmpeg_image *info,
 	return data;
 }
 
+static void *ffmpeg_image_orient(struct ffmpeg_image *info, void *in_data,
+				 int orient)
+{
+	const size_t sx = (size_t)info->cx;
+	const size_t sy = (size_t)info->cy;
+	uint8_t *data = NULL;
+
+	if (orient == 0 || orient == 1)
+		return in_data;
+
+	data = bmalloc(sx * 4 * sy);
+
+	if (orient >= 5 && orient < 9) {
+		info->cx = (int)sy;
+		info->cy = (int)sx;
+	}
+
+	uint8_t *src = in_data;
+
+	size_t off_dst;
+	size_t off_src = 0;
+	for (size_t x = 0; x < info->cx; x++) {
+		for (size_t y = 0; y < info->cy; y++) {
+			off_dst = (x + y * info->cx) * 4;
+
+			if (orient == 2) {
+				off_src = (sx - x - 1 + y * sx) * 4;
+			} else if (orient == 3) {
+				off_src = (sx - x - 1 + (sy - y - 1) * sx) * 4;
+			} else if (orient == 4) {
+				off_src = (x + (sy - y - 1) * sx) * 4;
+			} else if (orient == 5) {
+				off_src = (x * sx + y) * 4;
+			} else if (orient == 6) {
+				off_src = ((sy - x - 1) * sx + y) * 4;
+			} else if (orient == 7) {
+				off_src = (sx - y - 1 + (sy - x - 1) * sx) * 4;
+			} else if (orient == 8) {
```

Comment:
Are there any constants we can use instead of these magic numbers? If not, comments might be useful here.